src/utils/hooks/useSelectedColumns.ts

```diff
@@ -15,6 +15,13 @@ export const useSelectedColumns = <T extends {name: string}>(
         return settingsManager.readUserSettingsValue(storageKey, defaultColumnsIds) as string[];
     });
 
+    // Need to update selectedColumnsIds when passed storageKey changes.
+    React.useEffect(() => {
```

Comment:
If we do two separate components - we wont need useEffect

Reply:
And probably updating tables rendering will be smoother

Reply:
You can just call this hook twice for every storage key